src/cfdapi_transaction.cpp

```diff
@@ -143,42 +143,44 @@ ByteData TransactionApi::CreateSignatureHash(
       sighash_type);
 }
 
-ByteData TransactionApi::CreateSignatureHash(
+std::string TransactionApi::CreateSignatureHash(
     const std::string& tx_hex, const Txid& txid, uint32_t vout,
     const ByteData& key_data, const Amount& amount, HashType hash_type,
     const SigHashType& sighash_type) const {
-  std::string sig_hash;
+  ByteData sig_hash;
   int64_t amount_value = amount.GetSatoshiValue();
   TransactionController txc(tx_hex);
+  WitnessVersion version = WitnessVersion::kVersionNone;
+  if ((hash_type == HashType::kP2wpkh) || (hash_type == HashType::kP2wsh)) {
+    version = WitnessVersion::kVersion0;
+  }
 
-  if (hash_type == HashType::kP2pkh) {
-    sig_hash = txc.CreateP2pkhSignatureHash(
-        txid, vout,  // vout
-        Pubkey(key_data), sighash_type);
-  } else if (hash_type == HashType::kP2sh) {
-    sig_hash = txc.CreateP2shSignatureHash(
-        txid, vout, Script(key_data), sighash_type);
-  } else if (hash_type == HashType::kP2wpkh) {
-    sig_hash = txc.CreateP2wpkhSignatureHash(
-        txid, vout, Pubkey(key_data), sighash_type,
-        Amount::CreateBySatoshiAmount(amount_value));
-  } else if (hash_type == HashType::kP2wsh) {
-    sig_hash = txc.CreateP2wshSignatureHash(
-        txid, vout, Script(key_data), sighash_type,
-        Amount::CreateBySatoshiAmount(amount_value));
-  } else {
-    warn(
-        CFD_LOG_SOURCE,
-        "Failed to CreateSignatureHash. Invalid hash_type:  "
-        "hash_type={}",  // NOLINT
-        hash_type);
-    throw CfdException(
-        CfdError::kCfdIllegalArgumentError,
-        "Invalid hash_type. hash_type must be \"p2pkh\"(0) "
-        "or \"p2sh\"(1) or \"p2wpkh\"(2) or \"p2wsh\"(3).");  // NOLINT
+  switch (hash_type) {
+    case HashType::kP2pkh:
+      // fall-through
+    case HashType::kP2wpkh:
+      sig_hash = txc.CreateSignatureHash(
+          txid, vout, Pubkey(key_data), sighash_type, amount, version);
+      break;
+    case HashType::kP2sh:
+      // fall-through
+    case HashType::kP2wsh:
+      sig_hash = txc.CreateSignatureHash(
+          txid, vout, Script(key_data), sighash_type, amount, version);
+      break;
```

Comment:
正直どちらでも良いので好みの問題ですが、以下でいかがでしょうか？
```suggestion
  switch (hash_type) {
    case HashType::kP2wpkh:
      version = WitnessVersion::kVersion0;
      // fall-through
    case HashType::kP2pkh:    
      sig_hash = txc.CreateSignatureHash(
        txid, vout, Pubkey(key_data), sighash_type, amount, version);
      break;
    case HashType::kP2wsh:
      version = WitnessVersion::kVersion0;
      // fall-through
    case HashType::kP2sh:
      sig_hash = txc.CreateSignatureHash(
          txid, vout, Script(key_data), sighash_type, amount, version);
      break;
```

取り込まない場合、コメントだけつけてResolveしていただいて大丈夫です

Reply:
switch文の場合、breakなしで実処理（コメント以外）を書くと、たまにコンパイラやチェックツールでwarn指摘してくるのがあるので、あえて別判定としています。
（本格的に判定必要な場合、switch使わずif-elseifにしている箇所あり）